lib/vm/src/cache.rs

```diff
@@ -148,6 +167,37 @@ mod test {
         }
     }
 
+    #[test]
+    fn finds_cached_module() {
+        let tmp_dir = TempDir::new().unwrap();
+        let mut cache = unsafe { CosmCache::new(tmp_dir.path(), 10).unwrap() };
+        let id = cache.save_wasm(CONTRACT_0_7).unwrap();
+        let deps = dependencies(20);
+        let _instance = cache.get_instance(&id, deps).unwrap();
+        assert_eq!(cache.stats.hits_instance, 0);
+        assert_eq!(cache.stats.hits_module, 1);
+        assert_eq!(cache.stats.misses, 0);
+    }
+
+    #[test]
+    fn finds_cached_instance() {
+        let tmp_dir = TempDir::new().unwrap();
+        let mut cache = unsafe { CosmCache::new(tmp_dir.path(), 10).unwrap() };
+        let id = cache.save_wasm(CONTRACT_0_7).unwrap();
+        let deps1 = dependencies(20);
+        let deps2 = dependencies(20);
+        let deps3 = dependencies(20);
+        let instance1 = cache.get_instance(&id, deps1).unwrap();
+        cache.store_instance(&id, instance1);
+        let instance2 = cache.get_instance(&id, deps2).unwrap();
+        cache.store_instance(&id, instance2);
+        let instance3 = cache.get_instance(&id, deps3).unwrap();
+        cache.store_instance(&id, instance3);
+        assert_eq!(cache.stats.hits_instance, 2);
```

Comment:
Loving these stats. Not just for performance measurements, but to ensure the code is doing what we want.